pkg/controllers/controllers.go

```diff
@@ -0,0 +1,180 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/k3s-io/helm-controller/pkg/controllers/chart"
+	"github.com/k3s-io/helm-controller/pkg/controllers/common"
+	"github.com/k3s-io/helm-controller/pkg/generated/controllers/helm.cattle.io"
+	v1 "github.com/k3s-io/helm-controller/pkg/generated/controllers/helm.cattle.io/v1"
+	"github.com/rancher/lasso/pkg/cache"
+	"github.com/rancher/lasso/pkg/client"
+	"github.com/rancher/lasso/pkg/controller"
+	"github.com/rancher/wrangler/pkg/apply"
+	"github.com/rancher/wrangler/pkg/generated/controllers/batch"
+	batchcontrollers "github.com/rancher/wrangler/pkg/generated/controllers/batch/v1"
+	"github.com/rancher/wrangler/pkg/generated/controllers/core"
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/rancher/wrangler/pkg/generated/controllers/rbac"
+	rbaccontrollers "github.com/rancher/wrangler/pkg/generated/controllers/rbac/v1"
+	"github.com/rancher/wrangler/pkg/generic"
+	"github.com/rancher/wrangler/pkg/leader"
+	"github.com/rancher/wrangler/pkg/ratelimit"
+	"github.com/rancher/wrangler/pkg/start"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+type appContext struct {
+	v1.Interface
+
+	K8s   kubernetes.Interface
+	Core  corecontrollers.Interface
+	RBAC  rbaccontrollers.Interface
+	Batch batchcontrollers.Interface
+
+	Apply apply.Apply
+
+	ClientConfig clientcmd.ClientConfig
+	starters     []start.Starter
+}
+
+func (a *appContext) start(ctx context.Context) error {
+	return start.All(ctx, 50, a.starters...)
+}
+
+func Register(ctx context.Context, systemNamespace string, cfg clientcmd.ClientConfig, opts common.Options) error {
+	if len(systemNamespace) == 0 {
+		return errors.New("cannot start controllers on system namespace: system namespace not provided")
+	}
```

Comment:
We throw an error here if `systemNamespace` is empty, but down below at https://github.com/k3s-io/helm-controller/pull/157/files#diff-b32dd54d2bd7ecc467b19203287df80c4c1e3934ea80602aadd6658306b5cca6R73 it looks like the intent was to allow for watching all namespaces if the string is empty. Is this an error, or should that subsequent check be removed?